builder/builder.go

```
@@ -58,81 +58,115 @@ func getAuthData(filename string) (*dockerAuthData, error) {
+	var tagName string
+	if assignmentEnv {
+		tagName = fmt.Sprintf("%s/%s", imgBuilder.authData.Username, imgBuilder.LanguageImageFormat)
+	} else {
+		tagName = fmt.Sprintf("%s/%s:%s", imgBuilder.authData.Username,
+			imgBuilder.authData.Repository, imgBuilder.authData.Version)
+	}
+
+	return tagName
+}
+
+// Get Docker Build Context Tar Reader for building image.
+func getDockerBuildContextTar(dockerFilepath string) (*os.File, error) {
 	dockerFileReader, err := os.Open(dockerFilepath)
 	if err != nil {
 		log.Fatalf(" unable to open dockerfile: %v", err)
 	}
-	readDockerFile, err := ioutil.ReadAll(dockerFileReader)
+	fileInfo, err := os.Stat(dockerFilepath)
 	if err != nil {
-		log.Fatalf("unable to read dockerfile: %v", err)
+		return nil, err
 	}
 
-	tarHeader := &tar.Header{
-		Name: dockerFilepath,
-		Size: int64(len(readDockerFile)),
-	}
-	err = writer.WriteHeader(tarHeader)
+	buildContextTar := new(archivex.TarFile)
+	err = buildContextTar.Create(constants.BuildContextTar)
 	if err != nil {
-		log.Fatalf("unable to write tar header: %v", err)
+		return nil, err
 	}
-	_, err = writer.Write(readDockerFile)
+	err = buildContextTar.AddAll(constants.InstallationScriptsDir, true)
 	if err != nil {
-		log.Fatalf("unable to write tar body: %v", err)
+		return nil, err
 	}
-
-	// Use the tar of the Dockerfile while building image.
-	dockerFileTarReader := bytes.NewReader(tarBuffer.Bytes())
-
-	response, err := dockerClient.ImageBuild(
-		buildContext,
-		dockerFileTarReader,
-		types.ImageBuildOptions{
-			Dockerfile: dockerFilepath,
-			Tags: []string{fmt.Sprintf("%s/%s:%s", imgBuilder.authData.Username,
-				imgBuilder.authData.Repository, imgBuilder.authData.Version)}})
+	err = buildContextTar.Add(dockerFilepath, dockerFileReader, fileInfo)
 	if err != nil {
-		log.Fatalf("unable to build docker image: %v", err)
+		return nil, err
 	}
+
 	defer func() {
-		err = response.Body.Close()
+		err = buildContextTar.Close()
 		if err != nil {
 			log.Fatal(err)
```

Why not return the error here like done in other places?

---

Fixed